src/linalg/dense.jl

```diff
@@ -37,3 +37,46 @@ dense2arg = Dict{Symbol,Any}(
 # cond
 # sylvester
 # lyap
+
+@primitive chol(x),dy,y   chol_back(y, dy)
+
+# ref: formulua in https://arxiv.org/pdf/1602.07527.pdf
+# as described in https://arxiv.org/pdf/1710.08717.pdf
+# In julia L is upper triangular
+function chol_back(L, dL)
+    dL = triu(dL)
+    iL = inv(L)
+    S = iL * Symmetric(L*dL',:L) * iL'
+    S/2
+end
+
```

Comment:
@CarloLucibello let's not break the AutoGrad's regular testing pipeline.  Could you pls. add
1.  ```addtest(:chol, rand(3,3))```
2. I think we need to find a way to write similar tests for qr and lq and remove them from the test/primitives.jl

Reply:
I'd like to keep tests this way. It is kind of atypical in the julia ecosystem to add tests into the src/ folder, and I don't want to scratch my head to figure out how to add tests for `qr` in the non-standard way while the easy way is also the standard way